node/src/proxy_server/http_protocol_pack.rs

```diff
@@ -220,29 +224,30 @@ mod tests {
 
         let host = HttpProtocolPack {}.find_host(&data).unwrap();
 
-        assert_eq!(String::from("top.host.com"), host.name);
+        assert_eq!(host, Host::new("top.host.com", HTTP_PORT));
+        assert_eq!(HTTP_PORT, host.port);
     }
 
     #[test]
     fn returns_host_name_from_url_when_no_scheme() {
         let data = PlainData::new(
-            b"CONNECT good.url.dude/path.html HTTP/1.1\r\nHost: wrong.url.dude\r\n\r\n",
+            b"GET wrong.url.dude/path.html HTTP/1.1\r\nHost: good.url.dude\r\n\r\n",
```

Comment:
`CONNECT`s are unusual in the way their URL parameters are formed. The code under test doesn't care about that, but I still don't like a test with a manifestly incorrect request syntax, if I can avoid it. And I can. So I did.